### PR TITLE
H-1767: Write service to insert embeddings into Graph

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

When an entity is created/updated an embedding should be added to the database. As we use Ada (`text-embedding-ada-002`) to generate embeddings, which is an external service from OpenAI with latency considerations, this is done in a Temporal service.

## 🚫 Blocked by

- #3844 

## 🔍 What does this change?

- Adds a workflow to create embeddings
- Does not call the workflow, yet, this will be done in a follow-up

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph
## 🐾 Next steps

- H-1840
- H-1841

### CODE REVIEW

File: apps/hash-ai-worker-ts/src/activities/embeddings.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      },
+    };
+  }
+
+  // sort property types by their base url
+  params.propertyTypes.sort((a, b) =>
+    a.metadata.recordId.baseUrl.localeCompare(b.metadata.recordId.baseUrl),
+  );
+
+  const input = [""];
```
Comment (CiaranMn) [inline apps/hash-ai-worker-ts/src/activities/embeddings.ts:46]: I think we could make it easier to understand what's going on here without needing to know how it works already. Either via splitting out the variables, e.g.
```ts
  const propertyEmbeddings = [];
  let combinedEntityEmbedding = "";
  for (const propertyType of params.propertyTypes) {
    ...
    combinedEntityEmbedding += `${embeddingInput}\n`;
    propertyEmbeddings.push(embeddingInput);
  }

  const response = await openai.embeddings.create({
    input: [combinedEntityEmbedding, ...propertyEmbeddings],
```

or adding a comment, something like
```suggestion
  /**
   * We want to create embeddings for:
   * 1. Each individual '[Property Title]: [Value]' pair, and
   * 2. A list of all property key:value pairs
   * 
   * We use the first item in the array to store the combined 'all properties' list.
   */
  const input = [""];
```

File: apps/hash-ai-worker-ts/src/activities/graph.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}) => ({
+  async getEntityTypesByQuery(params: {
+    authentication: {
+      actorId: AccountId;
+    };
+    query: EntityTypeStructuralQuery;
+  }): Promise<Subgraph<EntityTypeRootType>> {
+    return graphApiClient
+      .getEntityTypesByQuery(params.authentication.actorId, params.query)
+      .then((response) => response.data as Subgraph<EntityTypeRootType>);
```
Comment (CiaranMn) [inline apps/hash-ai-worker-ts/src/activities/graph.ts:27]: A bit safer if the return type of `getEntityTypesByQuery` ever changes
```suggestion
      .then((response) => mapGraphApiSubgraphToSubgraph(response.data));
```

above
```ts
import {
  getPropertyTypes,
  mapGraphApiSubgraphToSubgraph,
} from "@local/hash-subgraph/stdlib";
```

**Review by CiaranMn — COMMENTED** [review]
Thanks @TimDiekmann, a few small comments

File: apps/hash-ai-worker-ts/package.json [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "@local/advanced-types": "0.0.0-private",
     "@local/hash-backend-utils": "0.0.0-private",
     "@local/hash-isomorphic-utils": "0.0.0-private",
+    "@local/hash-subgraph": "0.0.0-private",
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-ai-worker-ts/package.json:41]: We import functions from `hash-subgraph` so it has to be a runtime dependency.

File: apps/hash-ai-worker-ts/src/activities.ts [hunk 4]
```diff
@@ -3,10 +3,19 @@ import type {
   InferEntitiesCallerParams,
   InferEntitiesReturn,
 } from "@local/hash-isomorphic-utils/ai-inference-types";
+import {
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-ai-worker-ts/src/activities.ts:6]: It's fine to use `type` here but we still need the package as a runtime package.

File: apps/hash-ai-worker-ts/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-ai-worker-ts/package.json:41]: Oh, right, I just looked at the diff – wonder why this wasn't flagged before

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-ai-worker-ts/package.json:41]: I don't know if this makes a difference but previously we only had to import from `@local/hash-subgraph`. Now we need to import two functions from `@local/hash-subgraph/stdlib`. Could this have an impact?

[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-ai-worker-ts/package.json:41]: Shouldn't do, no. Can't find any exceptions to the rule (`no-extraneous-dependencies`) which would apply

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**